### PR TITLE
feat!: Add support for feature flag

### PR DESCRIPTION
**Pull request by mlnrt**
## Description
- Add support for `type` argument 
- Bump minimum supported version of Terraform and AWS provider to 1.0 and 5.0 respectively
- Update default output value to `null`

## Motivation and Context
The current version does not allow the configuration of AppConfig hosted Feature Flags

## Breaking Changes
- Yes - minimum support version bump

## How Has This Been Tested?
A new Feature Flag example has been created with its configuration set in `examples/_configs/feature_flag.json`

### CODE REVIEW

File: examples/feature_flag/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Feature Flag AWS AppConfig Example
```
Comment (bryantbiggs) [inline examples/feature_flag/README.md:1]: directory name should be `hosted`, not `appconfig_hosted`

File: variables.tf [hunk 2]
```diff
@@ -43,6 +43,12 @@ variable "config_profile_description" {
   default     = null
 }
 
+variable "config_profile_type" {
+  description = "Type of configurations contained in the profile. Valid values: AWS.AppConfig.FeatureFlags and AWS.Freeform"
+  type        = string
+  default     = "AWS.Freeform"
```
Comment (bryantbiggs) [inline variables.tf]:
```suggestion
  default     = null
```

File: examples/feature_flag/main.tf [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+################################################################################
+
+module "deactivated_appconfig" {
+  source = "../../"
+
+  name   = local.name
+  create = false
+}
+
+module "appconfig" {
```
Comment (bryantbiggs) [inline examples/feature_flag/main.tf]: we don't need another directory for this. just show this implementation in the `appconfig_hosted` example with its outputs

File: examples/_configs/feature_flag.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (bryantbiggs) [inline examples/_configs/feature_flag.json]: just implement this directly in the example, no need for separate directory

File: versions.tf [hunk 5]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 3.55"
+      version = ">= 4.6"
```
Comment (bryantbiggs) [inline versions.tf]: since this is a breaking change, lets update the min provider version to `5.0` and min Terraform version to `1.0`

**Comment by bryantbiggs** [comment]
@mlnrt are you still interested in carrying these changes forward?

**Comment by mlnrt** [comment]
@bryantbiggs sorry I have been very busy the past two months and will stay busy for few weeks. Can this wait a little more?

**Review by bryantbiggs — APPROVED** [review]
thank you!

**Comment by antonbabenko** [comment]
This PR is included in version 2.0.0 :tada:

**Pull request merged**